### PR TITLE
Optimize BlockAccount

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/1841

### CODE REVIEW

File: src/neo/SmartContract/Native/PolicyContract.cs [hunk 1]
```diff
@@ -71,6 +71,29 @@ public UInt160[] GetBlockedAccounts(StoreView snapshot)
...
+
+            var blockedList = snapshot.Storages.TryGet(CreateStorageKey(Prefix_BlockedAccounts))
+                ?.GetSerializableList<UInt160>().ToArray()
+                ?? Array.Empty<UInt160>();
+
+            if (blockedList.Length == 0) return false;
+
+            foreach (var acc in hashes)
+            {
+                foreach (var blockAccount in blockedList)
```
Comment (roman-khimov) [inline src/neo/SmartContract/Native/PolicyContract.cs:86]: Maybe some `Array.BinarySearch()`?

Comment (shargon) [reply src/neo/SmartContract/Native/PolicyContract.cs]: Done

File: src/neo/SmartContract/Native/PolicyContract.cs [hunk 2]
```diff
@@ -79,15 +79,11 @@ public bool IsAnyAccountBlocked(StoreView snapshot, params UInt160[] hashes)
...
-            foreach (var acc in hashes)
+            if (blockedList.Length > 0)
             {
-                foreach (var blockAccount in blockedList)
+                foreach (var acc in hashes)
                 {
-                    var comp = blockAccount.CompareTo(acc);
-                    if (comp == 0) return true;
-                    if (comp > 0) break; // It's already sorted
+                    if (Array.BinarySearch(blockedList, acc) == 0) return true;
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/PolicyContract.cs:86]:
```suggestion
                    if (Array.BinarySearch(blockedList, acc) > 0) return true;
```

`Array.BinarySearch` returns the index.

Comment (shargon) [reply src/neo/SmartContract/Native/PolicyContract.cs]: Good catch! fixed

File: src/neo/SmartContract/Native/PolicyContract.cs [hunk 3]
```diff
@@ -71,6 +71,25 @@ public UInt160[] GetBlockedAccounts(StoreView snapshot)
...
+
+            var blockedList = snapshot.Storages.TryGet(CreateStorageKey(Prefix_BlockedAccounts))
+                ?.GetSerializableList<UInt160>().ToArray()
+                ?? Array.Empty<UInt160>();
+
+            if (blockedList.Length > 0)
+            {
+                foreach (var acc in hashes)
+                {
+                    if (Array.BinarySearch(blockedList, acc) != -1) return true;
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/PolicyContract.cs:86]:
```suggestion
                    if (Array.BinarySearch(blockedList, acc) >= 0) return true;
```

The return value of `Array.BinarySearch` can be less then `-1`.

**Pull request merged**